### PR TITLE
fixed terra tests and new tests

### PR DESCRIPTION
**Pull request by khaledyoussef24**
provided new tests and fixed the old tests to the latest updates

#368

### CODE REVIEW

File: tests/mattermost/main.tf [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  provider "grid" {
+  }
+
+  # this data source is used to break circular dependency in cases similar to the following:
+  # vm: needs to know the domain in its init script
+  # gateway_name: needs the ip of the vm to use as backend.
+  # - the fqdn can be computed from grid_gateway_domain for the vm
+  # - the backend can reference the vm ip directly 
+  data "grid_gateway_domain" "domain" {
+    node = 2
```
Comment (mohamedamer453) [inline tests/mattermost/main.tf:26]: Change this node id with the id of a gateway node on testnet. same thing for the node id in line `59`

File: tests/peertube/main.tf [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  provider "grid" {
+  }
+
+  # this data source is used to break circular dependency in cases similar to the following:
+  # vm: needs to know the domain in its init script
+  # gateway_name: needs the ip of the vm to use as backend.
+  # - the fqdn can be computed from grid_gateway_domain for the vm
+  # - the backend can reference the vm ip directly 
+  data "grid_gateway_domain" "domain" {
+    node = 3
```
Comment (mohamedamer453) [inline tests/peertube/main.tf:23]: Change this node id with the id of a gateway node on testnet. same thing for the node id in line `61`

File: tests/taiga/main.tf [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    publicip = true
+  }
+}
+
+data "grid_gateway_domain" "domain" {
+  node = 2
+  name = "grid3taiga"
+}
+resource "grid_name_proxy" "p1" {
+  node            = 2
```
Comment (mohamedamer453) [inline tests/taiga/main.tf:73]: Change both node ids with the id of a gateway node on testnet.

**Review by mohamedamer453 — CHANGES_REQUESTED** [review]

**Comment by mohamedamer453** [comment]
Also some tests failed when i tried to run them.

- Peertube

   Test Failed with the following error:
   ```
   │ Error: couldn't generate deployments data: failed to assign node ips: invalid ip : invalid CIDR address: 
   ```

- taiga
   Test Failed with the following error:
   ```
   	taiga_test.go:64: 
                Error Trace:    taiga_test.go:64
                Error:          "superuser-init: Running\nufw: Running\nsshd: Running\ncontainerd: Blocked\nsshd-init: Success\ndockerd: Blocked\ntaiga: Blocked\nufw-init: Success\n" does not contain "containerd: Running"
   ```

File: tests/mattermost/mattermost_test.go [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+	assert.NotContains(t, string(out), "Destination Host Unreachable")
+
+	// Check that env variables set successfully
+	res, _ := tests.RemoteRun("root", ip, "cat /proc/1/environ")
+	assert.Contains(t, string(res), "TEST_VAR=this value for test")
+
+	// Check that the solution is running successfully
+
+	res1, _ := tests.RemoteRun("root", ip, "zinit list")
+	assert.Contains(t, res1, "mattermost: Running")
```
[the remark this answers is not in the document]
Comment (khaledyoussef24) [reply tests/mattermost/mattermost_test.go:63]: even when the service is working and built up it is running not success 

Screenshot from 2022-12-08 09-11-03

File: tests/taiga/taiga_test.go [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+	// out, _ := exec.Command("ping", ip, "-c 5", "-i 3", "-w 10").Output()
+	// assert.NotContains(t, string(out), "Destination Host Unreachable")
+
+	// Check that env variables set successfully
+	res, _ := tests.RemoteRun("root", ip, "cat /proc/1/environ")
+	assert.Contains(t, string(res), "TEST_VAR=this value for test")
+
+	res1, _ := tests.RemoteRun("root", ip, "zinit list")
+	assert.Contains(t, res1, "containerd: Running")
```
[the remark this answers is not in the document]
Comment (khaledyoussef24) [reply tests/taiga/taiga_test.go:64]: i will change it to taiga running

Screenshot from 2022-12-08 13-03-18

**Comment by mohamedamer453** [comment]
All of the requested changes were resolved except for one. please check it again.

I ran the tests but a couple failed with the following errors.

 - Presearch
   
   ```
    Error Trace:    presearch_test.go:67
                   Error:          "---\nsshkey: Success\nprerestorer: Success\nsshd: Running\npreupdater: \"Error(Exited(Pid(287), 125))\"\nprenode: \"Error(Exited(Pid(288), 125))\"\ndockerd: Running\n" does not contain "prenode: Success"
   ```

- kubernetes_with_k8s_module
```
    apply.go:15: 
                Error Trace:    apply.go:15
                                                        kubernetes_test.go:53
                Error:          Received unexpected error:
                                FatalError{Underlying: error while running command: exit status 1; There are some problems with the configuration, described below.
                            
                                The Terraform configuration must be valid before initialization so that
                                Terraform can determine which modules and providers need to be installed.
                                ╷
                                │ Error: Invalid validation error message
                                │ 
                                │   on .terraform/modules/kubernetes/modules/k8s-module/variables.tf line 33, in variable "master":
                                │   33:     error_message = "Memory must be at least 1GB"
                                │ 
                                │ The validation error message must be at least one full sentence starting
                                │ with an uppercase letter and ending with a period or question mark.
                                │ 
                                │ Your given message will be included as part of a larger Terraform error
                                │ message, written as English prose. For broadly-shared modules we suggest
                                │ using a similar writing style so that the overall result will be
                                │ consistent.
                                ╵
                            
                                ╷
                                │ Error: Invalid validation error message
                                │ 
                                │   on .terraform/modules/kubernetes/modules/k8s-module/variables.tf line 53, in variable "workers":
                                │   53:     error_message = "Memory must be at least 1GB"
                                │ 
                                │ The validation error message must be at least one full sentence starting
                                │ with an uppercase letter and ending with a period or question mark.
                                │ 
                                │ Your given message will be included as part of a larger Terraform error
                                │ message, written as English prose. For broadly-shared modules we suggest
                                │ using a similar writing style so that the overall result will be
                                │ consistent.
                                ╵
                            
                                ╷
                                │ Error: Invalid validation error message
                                │ 
                                │   on .terraform/modules/kubernetes/modules/k8s-module/validator/main.tf line 5, in variable "names":
                                │    5:     error_message = "Master and workers names must be distinct"
                                │ 
                                │ The validation error message must be at least one full sentence starting
                                │ with an uppercase letter and ending with a period or question mark.
                                │ 
                                │ Your given message will be included as part of a larger Terraform error
                                │ message, written as English prose. For broadly-shared modules we suggest
                                │ using a similar writing style so that the overall result will be
                                │ consistent.
                                ╵

```

- And the one i mentioned before wasn't fixed

> * singlenode_key_ed25519
>   This test also failed since it was not updated with the new format, you might need to remove [this line ](https://github.com/threefoldtech/terraform-provider-grid/blob/development/tests/singlenode_key_ed25519/main.tf#L29) from it like the rest of the tests that you updated.

**Comment by mohamedamer453** [comment]
All issues were fixed/resolved and the tests that were failing are passing now, the only issue left is the ` kubernetes_with_k8s_module`  with the same errors reported from last time, and after going over this error messages again i realized that they're not originating from the tests but from the module itself.




> * kubernetes_with_k8s_module
> 
> ```
>     apply.go:15: 
>                 Error Trace:    apply.go:15
>                                                         kubernetes_test.go:53
>                 Error:          Received unexpected error:
>                                 FatalError{Underlying: error while running command: exit status 1; There are some problems with the configuration, described below.
>                             
>                                 The Terraform configuration must be valid before initialization so that
>                                 Terraform can determine which modules and providers need to be installed.
>                                 ╷
>                                 │ Error: Invalid validation error message
>                                 │ 
>                                 │   on .terraform/modules/kubernetes/modules/k8s-module/variables.tf line 33, in variable "master":
>                                 │   33:     error_message = "Memory must be at least 1GB"
>                                 │ 
>                                 │ The validation error message must be at least one full sentence starting
>                                 │ with an uppercase letter and ending with a period or question mark.
>                                 │ 
>                                 │ Your given message will be included as part of a larger Terraform error
>                                 │ message, written as English prose. For broadly-shared modules we suggest
>                                 │ using a similar writing style so that the overall result will be
>                                 │ consistent.
>                                 ╵
>                             
>                                 ╷
>                                 │ Error: Invalid validation error message
>                                 │ 
>                                 │   on .terraform/modules/kubernetes/modules/k8s-module/variables.tf line 53, in variable "workers":
>                                 │   53:     error_message = "Memory must be at least 1GB"
>                                 │ 
>                                 │ The validation error message must be at least one full sentence starting
>                                 │ with an uppercase letter and ending with a period or question mark.
>                                 │ 
>                                 │ Your given message will be included as part of a larger Terraform error
>                                 │ message, written as English prose. For broadly-shared modules we suggest
>                                 │ using a similar writing style so that the overall result will be
>                                 │ consistent.
>                                 ╵
>                             
>                                 ╷
>                                 │ Error: Invalid validation error message
>                                 │ 
>                                 │   on .terraform/modules/kubernetes/modules/k8s-module/validator/main.tf line 5, in variable "names":
>                                 │    5:     error_message = "Master and workers names must be distinct"
>                                 │ 
>                                 │ The validation error message must be at least one full sentence starting
>                                 │ with an uppercase letter and ending with a period or question mark.
>                                 │ 
>                                 │ Your given message will be included as part of a larger Terraform error
>                                 │ message, written as English prose. For broadly-shared modules we suggest
>                                 │ using a similar writing style so that the overall result will be
>                                 │ consistent.
>                                 ╵
> ```

**Review by mohamedamer453 — APPROVED** [review]

**Pull request merged**